httpx/concurrency/base.py

```diff
@@ -44,20 +45,98 @@ class BaseTCPStream:
     backends, or for stand-alone test cases.
     """
 
+    def __init__(self, timeout: TimeoutConfig):
+        self.timeout = timeout
+        self.write_buffer = b""
```

Comment:
Slightly confusing thing here is that now both the `BaseTCPStream` and the Trio `TCPStream` have a `write_buffer`. Is the logic just duplicated in the Trio `TCPStream`?

Reply:
Ah, very good point! I seem to have forgotten to clean up the implementation from the trio backend. Will do. Thanks :)